### ISSUE TITLE
Allow Splunk/Splunkforwarder to be installed with "package" resource

### ISSUE THREAD

**Issue by inspired** [voxpupuli/puppet-splunk#125]
This is an enhancement request.
For those using Satellite or a package manager with the splunkforwarder in their repos it would be great to be able to install the splunkforwarder with the package resource instead of bundling it with the module/in Puppet repo.
Any reason this is not supported?

**Comment by bastelfreak** [comment]
hi @inspired, I'm not 100% sure what you mean. The package is currenty installed via package resource. Do you want to manage it outside of this module?

**Comment by inspired** [comment]
@bastelfreak Yeah, I'd like to install it with a RPM or APT repo instead of bundling it with the module. As far as I can see that's not supported? Correct me if I'm wrong :-)

**Comment by arjenz** [comment]
I'd like this feature too (was about to implement it myself, but haven't gotten around to it). Where possible, I try to add the Splunk packages to an internal repo, so yum and apt are able to just "simply" install them without puppet providing the binary package as a file.

**Comment by wilrodriguez** [comment]
If we could even just add a flag for this class to simply not manage the package that would even solve the problem for me.

**Comment by anthonysomerset** [comment]
if you set source to '' in the splunk class declaration, then it will assume to use package repos, we do this already internally and it works well for us

**Comment by jstraw** [comment]
I would even be ok with the puppet module package installing the splunkforwarder package, I have requirements that all my packages be managed centrally in a repository owned by a compliance team.  So our firewalls would block access to download the rpms in other ways, but there's an existing repo with splunkforwarder available.

**Comment by DLeich** [comment]
I'm going to add my +1 to this request.  Storing binaries and artifacts alongside source code in git is a really bad idea.  It's highly inefficient for git and wastes storage storage unnecessarily.  I am confused why we're not suggesting the use of the alternative source mechanism by default?
https://github.com/voxpupuli/puppet-splunk/blob/master/manifests/forwarder.pp#L49-L54

My understanding from reading the code is if you set $manage_package_source to false, and then provide a url to $package_source, then this should be doable.
